test/test_SSLContext.py

```diff
@@ -47,3 +58,10 @@ def test_HTTPConnection_with_custom_context(self):
         assert conn.ssl_context.check_hostname == True
         assert conn.ssl_context.verify_mode == ssl.CERT_REQUIRED
         assert conn.ssl_context.options & ssl.OP_NO_COMPRESSION != 0
+
+
+    def test_client_certificates(self):
+        context = hyper.tls.init_context(
+            cert=(CLIENT_CERT_FILE, CLIENT_KEY_FILE),
+            cert_password=b'abc123')
+        context = hyper.tls.init_context(cert=CLIENT_PEM_FILE)
```

Comment:
So this test doesn't actually *use* the certificates, and neither does the one below. That means you haven't yet noticed that there's a bug in the PyOpenSSL compatibility layer (or at least in the way you've understood it).

However, you should let me know if you think you don't have the time or expertise to write an appropriate test for this: I'm happy for you to just make the stylistic changes I've noted elsewhere and I'll extend the tests appropriately.